### ISSUE TITLE
Zurg running OOM while scanning the Plex TV Shows library

### ISSUE THREAD

**Issue by JurajNyiri** [debridmediamanager/zurg-testing#7]
### Description

While scanning my Plex TV shows library, zurg crashes often. When I mount webdav, plex server gets stuck and eventually marks everything as deleted. When I mount http, it does not get stuck, but since zurg crashes, it marks everything following the crash as deleted.

There doesn't seem to be any error message nor a pattern I can see in the logs.

### Logs example

```
2023-11-24T13:30:24.854Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:39:42.198Z	WARN	file	File ... is no longer available
2023-11-24T13:39:43.535Z	WARN	file	File ... is not yet available, zurg is repairing the torrent
2023-11-24T13:39:46.358Z	WARN	file	File ... is not yet available, zurg is repairing the torrent
2023-11-24T13:39:46.930Z	WARN	file	File ... is not yet available, zurg is repairing the torrent
2023-11-24T13:39:53.040Z	WARN	file	File ... is no longer available
2023-11-24T13:39:53.677Z	WARN	file	File ... is not yet available, zurg is repairing the torrent
2023-11-24T13:39:54.281Z	WARN	file	File ... is not yet available, zurg is repairing the torrent
2023-11-24T13:39:56.051Z	INFO	manager	Detected changes! Refreshing 5554 torrents
2023-11-24T13:39:56.430Z	DEBUG	realdebrid	Got info for torrent ... (progress=100%)
2023-11-24T13:39:56.430Z	WARN	manager	Torrent id=... is partly expired, it has 10 selected files but only 0 links
2023-11-24T13:39:56.434Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:39:57.386Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:39:57.407Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:39:57.420Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:39:57.584Z	INFO	manager	Fetched info for 5554 torrents
2023-11-24T13:40:00.197Z	DEBUG	config	Loading config file ./config.yml
2023-11-24T13:40:00.200Z	DEBUG	config	Detected config version: v1
2023-11-24T13:40:11.943Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:40:11.946Z	DEBUG	realdebrid	Got info for torrent ... (progress=100%)
2023-11-24T13:40:11.946Z	WARN	manager	Torrent id=... is partly expired, it has 10 selected files but only 0 links
2023-11-24T13:40:12.468Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:40:12.487Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:40:12.510Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:40:12.994Z	INFO	manager	Fetched info for 5554 torrents
2023-11-24T13:40:13.141Z	INFO	manager	Compiled into 4077 torrents, 0 were missing info
2023-11-24T13:40:13.213Z	INFO	manager	Checking for torrents to repair
2023-11-24T13:40:13.285Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-24T13:40:13.288Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-24T13:40:13.289Z	INFO	manager	Finished checking for torrents to repair
2023-11-24T13:40:13.289Z	INFO	manager	Starting periodic refresh
2023-11-24T13:40:13.290Z	INFO	zurg	Starting server on [::]:9999
...
2023-11-24T13:42:30.816Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-24T13:42:32.508Z	DEBUG	config	Loading config file ./config.yml
...
2023-11-24T13:45:02.647Z	WARN	file	... not yet available, zurg is repairing the torrent
2023-11-24T13:45:04.693Z	DEBUG	config	Loading config file ./config.yml
...
2023-11-24T13:45:46.428Z	INFO	manager	Finished checking for torrents to repair
2023-11-24T13:45:46.435Z	INFO	hooks	Output of hook on_library_update:
hook
...
2023-11-24T13:46:17.882Z	DEBUG	config	Loading config file ./config.yml
2023-11-24T13:46:30.909Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:54:31.335Z	INFO	manager	Detected changes! Refreshing 5554 torrents
2023-11-24T13:54:31.742Z	DEBUG	realdebrid	Got info for torrent ... (progress=100%)
2023-11-24T13:54:31.744Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:54:32.084Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:54:32.531Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:54:32.982Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T13:54:33.152Z	INFO	manager	Fetched info for 5554 torrents
2023-11-24T13:54:34.778Z	DEBUG	config	Loading config file ./config.yml
...
2023-11-24T14:05:08.470Z	WARN	file	File ... is no longer available
2023-11-24T14:05:10.668Z	DEBUG	config	Loading config file ./config.yml
...
2023-11-24T14:05:56.779Z	WARN	file	File ... is not yet available, zurg is repairing the torrent
2023-11-24T14:06:19.058Z	DEBUG	config	Loading config file ./config.yml
```

### Frequency of restarts

> docker logs zurg0.9 2>&1 | grep "Starting server"

```
2023-11-24T12:13:34.316Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T12:14:20.592Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T12:27:25.781Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T12:31:27.199Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T12:43:13.100Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T12:44:02.413Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T12:44:48.954Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T12:46:16.715Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T12:47:14.742Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:02:00.483Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:04:23.828Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:07:22.423Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:11:47.232Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:19:56.929Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:22:44.023Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:26:36.159Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:27:53.100Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:30:24.854Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:40:13.290Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:41:15.712Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:42:45.842Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:43:46.949Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:45:17.918Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T13:46:30.909Z	INFO	zurg	Starting server on [::]:9999
```

### Environment

I am running on Mac Mini M1. Using http endpoint for zurg.

This is how I launch zurg:

```
docker run -d \
  --name zurg0.9 \
  -p 9999:9999 \
  -v /Users/jurajnyiri/repos/zurg-testing/config.yml:/app/config.yml \
  -v /Users/jurajnyiri/repos/zurg-testing/zurgdata:/app/data \
  --restart unless-stopped \
  ghcr.io/debridmediamanager/zurg-testing:v0.9.0
```

```
docker --version
Docker version 24.0.6, build ed223bc
```
This is how I mount rclone:
```
/Users/jurajnyiri/rclone mount -o modules=iconv,from_code=UTF-8 --config=/Users/jurajnyiri/.config/rclone/rclone.conf \
        --allow-other \
        --poll-interval 15s \
        --log-level=INFO \
        --log-file="$media"rclone-mount.log \
        zurg: "${mntpoint}/Zurg"
```

Previously I had these as well but removed them for debugging purposes:

```
        --dir-cache-time 10s \
        --cache-dir="${media}/tmp/rclone_cache_Zurg" \
        --vfs-cache-mode full \
        --vfs-cache-max-size 16G \
        --vfs-cache-max-age=24h \
```

```
/Users/jurajnyiri/rclone --version
rclone v1.64.2
- os/version: darwin 13.6 (64 bit)
- os/kernel: 22.6.0 (arm64)
- os/type: darwin
- os/arch: arm64 (ARMv8 compatible)
- go/version: go1.21.3
- go/linking: dynamic
- go/tags: cmount
```

This is what I see in rclone log when zurg goes down multiple times during the period:

```
2023/11/24 14:43:39 ERROR : IO error: error listing ...: read tcp 192.168.100.19:63209->192.168.100.19:9999: read: connection reset by peer
2023/11/24 14:43:39 ERROR : ...: read tcp 192.168.100.19:63210->192.168.100.19:9999: read: connection reset by peer
...
2023/11/24 14:44:29 ERROR : ...: ReadFileHandle.Read error: low level retry 1/10: file already closed
2023/11/24 14:44:29 ERROR : ...: ReadFileHandle.Read error: low level retry 2/10: invalid seek position
2023/11/24 14:44:29 ERROR : ...: ReadFileHandle.Read error: low level retry 3/10: invalid seek position
2023/11/24 14:44:29 ERROR : ...: ReadFileHandle.Read error: low level retry 4/10: invalid seek position
2023/11/24 14:44:29 ERROR : ...: ReadFileHandle.Read error: low level retry 5/10: invalid seek position
2023/11/24 14:44:29 ERROR : ...: ReadFileHandle.Read error: low level retry 6/10: invalid seek position
2023/11/24 14:44:29 ERROR : ...: ReadFileHandle.Read error: low level retry 7/10: invalid seek position
2023/11/24 14:44:29 ERROR : ...: ReadFileHandle.Read error: low level retry 8/10: invalid seek position
2023/11/24 14:44:29 ERROR : ...: ReadFileHandle.Read error: low level retry 9/10: invalid seek position
2023/11/24 14:44:29 ERROR : ...: ReadFileHandle.Read error: low level retry 10/10: invalid seek position
...
2023/11/24 14:44:59 ERROR : ...mkv: ReadFileHandle.Read error: invalid seek position
2023/11/24 14:44:59 ERROR : IO error: invalid seek position
2023/11/24 14:44:59 ERROR : ...mkv: ReadFileHandle.Release error: file already closed
2023/11/24 14:44:59 ERROR : IO error: file already closed
...
2023/11/24 14:45:09 ERROR : IO error: Open failed: Get "http://192.168.100.19:9999/http/shows/....mkv": read tcp 192.168.100.19:65280->192.168.100.19:9999: read: connection reset by peer
...
2023/11/24 14:45:09 ERROR : IO error: Open failed: Get "http://192.168.100.19:9999/http/shows/....mkv": EOF
```

Rclone.conf:

```
[zurg]
type = http
url = http://192.168.100.19:9999/http
no_head = false
no_slash = false

[zurg-webdav]
type = webdav
url = http://192.168.100.19:9999
vendor = other
pacer_min_sleep = 0
```

Config.yml:

```
# Zurg configuration version
zurg: v1

token: ... # https://real-debrid.com/apitoken

host: "[::]" # do not change this if you are running it inside a docker container
port: 9999 # do not change this if you are running it inside a docker container
concurrent_workers: 200
check_for_changes_every_secs: 15
info_cache_time_hours: 12

enable_repair: true # BEWARE! THERE CAN ONLY BE 1 INSTANCE OF ZURG THAT SHOULD REPAIR YOUR TORRENTS
retain_folder_name_extension: false # if true, zurg won't modify the filenames from real-debrid
on_library_update: |
  echo "hook"

network_buffer_size: 1048576 # 1 MiB
preferred_hosts: # Run the script here https://github.com/debridmediamanager/real-debrid-network-test
  - 43.download.real-debrid.cloud
  - 69.download.real-debrid.cloud
  - 57.download.real-debrid.cloud
  - 22.download.real-debrid.cloud
  - 23.download.real-debrid.cloud
  - 55.download.real-debrid.cloud
  - 42.download.real-debrid.cloud
  - 61.download.real-debrid.cloud
  - 68.download.real-debrid.cloud
  - 62.download.real-debrid.cloud
  - 32.download.real-debrid.cloud
  - 66.download.real-debrid.cloud
  - 41.download.real-debrid.cloud
  - 53.download.real-debrid.cloud
# curl https://raw.githubusercontent.com/debridmediamanager/real-debrid-network-test/main/real-debrid-network-test.sh | bash

# List of directory definitions and their filtering rules
directories:
  shows:
    group: media
    group_order: 20
    filters:
      - has_episodes: true  # intelligent detection of episode files inside a torrent
      - regex: /season[\s\.]?\d/i          # Capture torrent names with the term 'season' in any case
      - regex: /saison[\s\.]?\d/i          # For non-English namings
      - regex: /stagione[\s\.]?\d/i        # if there's french, there should be italian too
      - regex: /s\d\d/i           # Capture common season notations like S01, S02, etc.
      - regex: /\btv/i            # anything that has TV in it is a TV show, right?
      - contains: complete
      - contains: seasons
      - any_file_inside_regex: /s\d\d/i
      - any_file_inside_regex: /\d\s?x\s?\d\d/i

  movies:
    group: media
    group_order: 30 
    filters:
      - regex: /.*/ # you cannot leave a directory without filters because it will not have any torrents in it
```

Please let me know how I can help debug further.

**Comment by JurajNyiri** [comment]
In docker ps status I see this: ```Up 2 minutes (health: starting)```.

On docker inspect I see:

```
"Health": {
                "Status": "starting",
                "FailingStreak": 3,
                "Log": [
                    {
                        "Start": "2023-11-24T15:48:02.225754875Z",
                        "End": "2023-11-24T15:48:02.3136985Z",
                        "ExitCode": 1,
                        "Output": "/app/healthcheck.sh: line 2: yaml: not found\nnc: missing port number\n"
                    },
                    {
                        "Start": "2023-11-24T15:49:02.319879292Z",
                        "End": "2023-11-24T15:49:02.391029417Z",
                        "ExitCode": 1,
                        "Output": "/app/healthcheck.sh: line 2: yaml: not found\nnc: missing port number\n"
                    },
                    {
                        "Start": "2023-11-24T15:51:07.506845878Z",
                        "End": "2023-11-24T15:51:07.604834669Z",
                        "ExitCode": 1,
                        "Output": "/app/healthcheck.sh: line 2: yaml: not found\nnc: missing port number\n"
                    },
                    {
                        "Start": "2023-11-24T15:52:07.741822086Z",
                        "End": "2023-11-24T15:52:07.839568003Z",
                        "ExitCode": 1,
                        "Output": "/app/healthcheck.sh: line 2: yaml: not found\nnc: missing port number\n"
                    },
                    {
                        "Start": "2023-11-24T15:53:07.913334128Z",
                        "End": "2023-11-24T15:53:08.015978545Z",
                        "ExitCode": 1,
                        "Output": "/app/healthcheck.sh: line 2: yaml: not found\nnc: missing port number\n"
                    }
                ]
            }
        }
```

**Comment by JurajNyiri** [comment]
I modified the file:

```
#!/bin/sh
exit 0
port=$(yaml read /app/config.yml port)
nc -z localhost $port || exit 1
```

and added chmod +x. Now the container reports `Up 3 minutes (healthy)`

**Comment by JurajNyiri** [comment]
Unfortunately, healthcheck was not the cause of restarts.

```
2023-11-24T16:01:30.879Z	INFO	zurg	Starting server on [::]:9999
2023-11-24T16:09:02.109Z	INFO	manager	Detected changes! Refreshing 5555 torrents
2023-11-24T16:09:03.064Z	DEBUG	realdebrid	Got info for torrent ... (progress=100%)
2023-11-24T16:09:03.069Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T16:09:03.075Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T16:09:03.081Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T16:09:03.086Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T16:09:03.333Z	INFO	manager	Fetched info for 5555 torrents
2023-11-24T16:09:03.651Z	INFO	manager	Compiled into 4077 torrents, 0 were missing info
2023-11-24T16:09:03.732Z	INFO	manager	Checking for torrents to repair
2023-11-24T16:09:03.809Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-24T16:09:03.810Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-24T16:09:03.813Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-24T16:09:03.815Z	INFO	manager	Finished checking for torrents to repair
2023-11-24T16:09:03.821Z	INFO	hooks	Output of hook on_library_update:
hook

2023-11-24T16:09:23.889Z	DEBUG	config	Loading config file ./config.yml
2023-11-24T16:09:23.890Z	DEBUG	config	Detected config version: v1
2023-11-24T16:09:35.837Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T16:09:36.053Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T16:09:36.383Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T16:09:36.602Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-24T16:09:36.612Z	INFO	manager	Fetched info for 5555 torrents
2023-11-24T16:09:36.806Z	INFO	manager	Compiled into 4077 torrents, 0 were missing info
2023-11-24T16:09:36.896Z	INFO	manager	Checking for torrents to repair
2023-11-24T16:09:36.976Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-24T16:09:36.979Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-24T16:09:36.981Z	INFO	manager	Finished checking for torrents to repair
2023-11-24T16:09:36.981Z	INFO	manager	Starting periodic refresh
2023-11-24T16:09:36.981Z	INFO	zurg	Starting server on [::]:9999
```

**Comment by yowmamasita** [comment]
I have a probably fix for this incoming on v0.9.1

**Comment by yowmamasita** [comment]
Healthcheck is a different issue and should not impact functionality

**Comment by JurajNyiri** [comment]
@yowmamasita thank you for the new release!

I have upgraded to 0.9.1 using the same config.yml file as described above and ran the scan for tv shows. Unfortunately the problem has not been resolved with the same behaviour as described above.

> docker logs zurg0.9.1

```
2023-11-25T16:55:04.432Z	DEBUG	config	Loading config file ./config.yml
2023-11-25T16:55:04.433Z	DEBUG	config	Detected config version: v1
2023-11-25T16:55:16.209Z	DEBUG	realdebrid	Got info for torrent ... (progress=49%)
2023-11-25T16:55:16.868Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T16:55:17.002Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T16:55:17.029Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T16:55:17.723Z	INFO	manager	Fetched info for 5560 torrents
2023-11-25T16:55:18.165Z	INFO	manager	Compiled into 4077 torrents, 0 were missing info
2023-11-25T16:55:18.250Z	INFO	manager	Checking for torrents to repair
2023-11-25T16:55:18.327Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-25T16:55:18.330Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-25T16:55:18.331Z	INFO	manager	Finished checking for torrents to repair
2023-11-25T16:55:18.331Z	INFO	zurg	Starting server on [::]:9999
2023-11-25T16:55:18.332Z	INFO	manager	Starting periodic refresh
2023-11-25T16:55:53.667Z	WARN	file	File was changed and is not streamable: ... and ...
2023-11-25T16:55:55.316Z	WARN	file	File was changed and is not streamable: ... and ...
2023-11-25T16:55:56.173Z	WARN	file	File was changed and is not streamable: ... and ...
2023-11-25T16:55:57.035Z	WARN	file	File was changed and is not streamable: ... and ...
2023-11-25T17:02:48.569Z	WARN	file	File /... is no longer available
2023-11-25T17:02:49.772Z	WARN	file	File ... is not yet available, zurg is repairing the torrent
2023-11-25T17:02:50.799Z	WARN	file	File ... is not yet available, zurg is repairing the torrent
2023-11-25T17:02:51.381Z	WARN	file	File ... is not yet available, zurg is repairing the torrent
2023-11-25T17:02:57.892Z	WARN	file	File /... is no longer available
2023-11-25T17:02:58.514Z	WARN	file	File ... is not yet available, zurg is repairing the torrent
2023-11-25T17:02:59.128Z	WARN	file	File ... is not yet available, zurg is repairing the torrent
2023-11-25T17:03:06.789Z	INFO	manager	Detected changes! Refreshing 5560 torrents
2023-11-25T17:03:07.088Z	DEBUG	realdebrid	Got info for torrent ... (progress=50%)
2023-11-25T17:03:07.202Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T17:03:08.434Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T17:03:08.679Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T17:03:11.001Z	DEBUG	config	Loading config file ./config.yml
2023-11-25T17:03:11.005Z	DEBUG	config	Detected config version: v1
2023-11-25T17:03:22.909Z	DEBUG	realdebrid	Got info for torrent ... (progress=50%)
2023-11-25T17:03:23.352Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T17:03:23.360Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T17:03:24.072Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T17:03:24.073Z	INFO	manager	Fetched info for 5560 torrents
2023-11-25T17:03:24.253Z	INFO	manager	Compiled into 4077 torrents, 0 were missing info
2023-11-25T17:03:24.332Z	INFO	manager	Checking for torrents to repair
2023-11-25T17:03:24.409Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-25T17:03:24.413Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-25T17:03:24.414Z	INFO	manager	Finished checking for torrents to repair
2023-11-25T17:03:24.415Z	INFO	manager	Starting periodic refresh
2023-11-25T17:03:24.437Z	INFO	zurg	Starting server on [::]:9999
2023-11-25T17:05:09.539Z	DEBUG	config	Loading config file ./config.yml
2023-11-25T17:05:09.546Z	DEBUG	config	Detected config version: v1
2023-11-25T17:05:21.301Z	DEBUG	realdebrid	Got info for torrent ... (progress=50%)
2023-11-25T17:05:21.355Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T17:05:22.527Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T17:05:22.528Z	DEBUG	realdebrid	Got info for torrent ... (progress=0%)
2023-11-25T17:05:22.857Z	INFO	manager	Fetched info for 5560 torrents
2023-11-25T17:05:23.066Z	INFO	manager	Compiled into 4077 torrents, 0 were missing info
2023-11-25T17:05:23.234Z	INFO	manager	Checking for torrents to repair
2023-11-25T17:05:23.304Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-25T17:05:23.307Z	DEBUG	manager	Skipping ... for repairs because it is in progress
2023-11-25T17:05:23.308Z	INFO	manager	Finished checking for torrents to repair
2023-11-25T17:05:23.308Z	INFO	manager	Starting periodic refresh
2023-11-25T17:05:23.309Z	INFO	zurg	Starting server on [::]:9999
```

What can I do to help?

**Comment by JurajNyiri** [comment]
As for the healthcheck, I am seeing unhealthy now:

```
"Log": [
                    {
                        "Start": "2023-11-25T17:14:11.952436758Z",
                        "End": "2023-11-25T17:14:12.092591967Z",
                        "ExitCode": 1,
                        "Output": ""
                    },
                    {
                        "Start": "2023-11-25T17:15:12.1193153Z",
                        "End": "2023-11-25T17:15:12.304317092Z",
                        "ExitCode": 1,
                        "Output": ""
                    },
                    {
                        "Start": "2023-11-25T17:16:12.314185717Z",
                        "End": "2023-11-25T17:16:12.498837717Z",
                        "ExitCode": 1,
                        "Output": ""
                    },
                    {
                        "Start": "2023-11-25T17:17:12.537940217Z",
                        "End": "2023-11-25T17:17:12.638424634Z",
                        "ExitCode": 1,
                        "Output": ""
                    },
                    {
                        "Start": "2023-11-25T17:18:36.187830131Z",
                        "End": "2023-11-25T17:18:36.300984922Z",
                        "ExitCode": 1,
                        "Output": ""
                    }
                ]
```

**Comment by yowmamasita** [comment]
@JurajNyiri apologies, there's something wrong with the health check again, I'm building the hotfix now

**Comment by JurajNyiri** [comment]
Thank you, I pulled the image again and now the healthcheck works. Is there anything I can do to help you help me debug the crashing issue?

**Comment by JurajNyiri** [comment]
I was able to find the cause of the restarts - docker engine was running out of memory. I had it set to 1GB swap and 1GB ram. I raised this to 4GB RAM and 4GB swap.

I started zurg and plex scan at the same time and monitored zurg memory usage via `docker stats`:
20:25: 266MB 
20:29: 480MB
20:33: 825MB
20:34: 1005MB
20:38: 1.22GB
20:40: 830MB
20:43: 1.17GB
20:44 1.3GB
20:47: 1GB
20:50: 1.5GB
20:54: 1.3GB
20:56: 1.06GB
20:58: 1.591GB
21:00: 1.389GB
21:04: 1.507GB
21:07: 1.17GB
21:09: 1.74GB
21:11: 1.16GB
21:13 1.398GB
21:14: 1.852GB
21:21: 1.544GB
21:23: 1.92GB
21:24: 2.03GB
21:26: 2.07GB 
21:27: 2.25GB
21:28: 2.44GB
21:31: 1.877GB
21:32: 2.2GB
